frontend/pages/admin/IntegrationsPage/cards/Mdm/components/RequestCSRModal/RequestCSRModal.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { FormEvent, useState, useContext, useEffect } from "react";
+
+import { AppContext } from "context/app";
+
+import { IRequestCSRFormData } from "interfaces/request_csr";
+
+import Button from "components/buttons/Button";
+// @ts-ignore
+import InputField from "components/forms/fields/InputField";
+import Spinner from "components/Spinner";
+import DataError from "components/DataError";
+import Icon from "components/Icon";
+import Modal from "components/Modal";
+
+const baseClass = "modal request-csr-modal";
+interface IRequestCSRModalProps {
+  onCancel: () => void;
+  testingConnection?: boolean;
+}
+
+interface IFormField {
+  name: string;
+  value: string;
+}
+
+const RequestCSRModal = ({
+  onCancel,
+  testingConnection,
+}: IRequestCSRModalProps): JSX.Element => {
+  const { currentUser, config } = useContext(AppContext);
+
+  const [formData, setFormData] = useState<IRequestCSRFormData>({
+    email: currentUser?.email || "",
+    orgName: config?.org_info?.org_name || "",
+  });
+
+  const [requestState, setRequestState] = useState<
+    "loading" | "error" | "success" | "invalid" | undefined
+  >(undefined);
+  const [invalidMessage, setInvalidMessage] = useState<string>("");
+
+  useEffect(() => {
+    requestState === "invalid"
+      ? setInvalidMessage("Email")
+      : setInvalidMessage("");
+  }, [requestState]);
+
+  const { email, orgName } = formData;
+
+  // destructure change event to its name and value
+  const onInputChange = ({ name, value }: IFormField) => {
+    setFormData({ ...formData, [name]: value });
+  };
+
+  const onFormSubmit = (evt: FormEvent): void => {
+    evt.preventDefault();
+
+    // TODO: once API is finished, change below to actually call it
+    setRequestState("loading");
+    setTimeout(() => setRequestState("success"), 1000);
+  };
+
+  const renderRequestCSRForm = () => {
+    switch (requestState) {
+      case "success":
```

Comment:
rendering with a switch statement for only three states is overly complicated. the control flow can easily be rewritten into something more like this.

```tsx
if (requestState === "success") return <SuccessMessage />
if (requestState === "error") return <DataError />

return (
<div>
  ... UI mockup for form ...
</div>)
```

this removes the need for the switch/case statements with tricky break statements and abstracts the success message mockup out somewhere else.